### PR TITLE
Canu cleanup commit WIP

### PR DESCRIPTION
**Pull request by mmiladi**

### CODE REVIEW

**Comment by mmiladi** [comment]
Please do not merge yet :)

File: tools/canu/canu.xml [hunk 1]
```diff
@@ -103,47 +103,61 @@
...
                 <param name="minReads" value="10" />
             </section>
-            <output name="corrected_reads" ftype="fasta.gz" decompress="True" file="canu_corrected_reads.fa.gz"/>
+            <output name="contigs" ftype="fasta" file="ecoli_canu_contigs_result3.fa"/>
+            <output name="unitigs" ftype="fasta" file="ecoli_canu_unitigs_result3.fa"/>
+            <output name="unassembled" ftype="fasta" file="ecoli_unassembled_result3.fa"/>
+            <output name="corrected_reads" ftype="fasta.gz" decompress="True" file="ecoli_canu_corrected_reads_result3.fa.gz"/>
+        </test>
+        <test>
+            <param name="inputs" ftype="fasta" value="ecoli-reads.fasta"/>
```
[the remark this answers is not in the document]
Comment (mmiladi) [reply tools/canu/canu.xml]: Planemo returned the weird database error. Outputs are fixed now, some of them could be empty depending on params.

File: tools/canu/canu.xml [hunk 2]
```diff
@@ -103,47 +103,61 @@
...
         </data>
-        <data name="unassembled" format="fasta" from_work_dir="out_dir/canu.unassembled.fasta" label="${tool.name} on ${on_string} (contigs)">
+        <data name="unassembled" format="fasta" from_work_dir="out_dir/canu.unassembled.fasta" label="${tool.name} on ${on_string} (unassembled)">
         </data>
         <data name="unitigs" format="fasta" from_work_dir="out_dir/canu.unitigs.fasta" label="${tool.name} on ${on_string} (unitigs)">
         </data>
-        <data name="corrected_reads" format="fasta.gz" from_work_dir="out_dir/canu.correctedReads.fasta.gz" label="${tool.name} on ${on_string} (unitigs)">
+        <data name="corrected_reads" format="fasta.gz" from_work_dir="out_dir/canu.correctedReads.fasta.gz" label="${tool.name} on ${on_string} (corrected reads)">
+        </data>
+        <data name="trimmed_reads" format="fasta.gz" from_work_dir="out_dir/canu.trimmedReads.fasta.gz" label="${tool.name} on ${on_string} (trimmed reads)">
```
[the remark this answers is not in the document]
Comment (mmiladi) [reply tools/canu/canu.xml]: Yes, Non-empty if -trim is set. :)

File: tools/canu/canu.xml [hunk 3]
```diff
@@ -102,48 +102,67 @@
...
+        </test>
+        <test expect_num_outputs="1">
+            <param name="inputs" ftype="fasta" value="ecoli-reads.fasta"/>
+            <param name="minReadLength" value="2000" />
+            <param name="stage" value="-trim"/>
+            <param name="genomeSize" value="4.6m" />
+            <section name="contigFilter">
+                <param name="minReads" value="10" />
+            </section>
+            <output name="trimmed_reads" ftype="fasta.gz" compare="sim_size" delta="20000" file="ecoli_canu_trimmed_reads_result4.fa.gz"/>
```
Comment (bgruening) [inline tools/canu/canu.xml]: maybe decompress as well?

**Comment by mmiladi** [comment]
This must be final final update :D

**Pull request merged**